src/tracks/playabletrack/wavetrack/ui/WaveformView.cpp

```diff
@@ -736,8 +737,7 @@ void DrawClipWaveform(TrackPanelDrawingContext &context, size_t channel,
 
    WaveDisplay display(hiddenMid.width);
 
-   const double pps =
-      averagePixelsPerSample * rate;
+   const double pps = averagePixelsPerSample * rate / stretchRatio;
```

Comment:
I have a doubt about the correctness.  Where did the `averagePixelsPerSample` value come from?  I found it in the function `CalculateAdjustmentForZoomLevel`.  That function uses clip rate but not stretch ratio.  So I think it computes an incorrect pixels per sample value.  I'm not sure what consequences follow.

Reply:
I reexamined just what is done with `ClipParameters::rate` and I see that for spectrograms, sometimes the rate is used for calculations relating to the y coordinate on the frequency scale -- and those should still use the unadjusted clip rate, but other calculations that convert sample counts to time points should be updated.

So I anticipate seeing this as I review later commits.